sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/InjectableDispatchersCheck.kt

```diff
@@ -0,0 +1,44 @@
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtQualifiedExpression
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.FilteredAbstractCheck
+import org.sonarsource.kotlin.api.FunMatcher
+import org.sonarsource.kotlin.api.predictRuntimeValueExpression
+import org.sonarsource.kotlin.api.resolveReferenceTarget
+import org.sonarsource.kotlin.api.secondaryOf
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private const val MESSAGE = "Avoid hardcoded dispatchers"
+private const val DISPATCHERS_OBJECT = "kotlinx.coroutines.Dispatchers"
+private const val COROUTINES_PACKAGE = "kotlinx.coroutines"
+
+private val FUNS_ACCEPTING_DISPATCHERS = listOf(
+    FunMatcher(qualifier = COROUTINES_PACKAGE, name = "withContext"),
+    FunMatcher(qualifier = COROUTINES_PACKAGE, name = "async"),
+    FunMatcher(qualifier = COROUTINES_PACKAGE, name = "launch"),
+)
+
+@Rule(key = "S6310")
+class InjectableDispatchersCheck : FilteredAbstractCheck() {
+    override fun functionsToVisit() = FUNS_ACCEPTING_DISPATCHERS
+
+    override fun visitCallFiltered(resolvedCall: ResolvedCall<*>, callExpression: KtCallExpression, kotlinFileContext: KotlinFileContext) {
+        val bindingContext = kotlinFileContext.bindingContext
+
+        val argExpr = resolvedCall.valueArgumentsByIndex?.get(0)?.arguments?.get(0)?.getArgumentExpression() ?: return
+        val argValueExpr = argExpr.predictRuntimeValueExpression(bindingContext) as? KtQualifiedExpression ?: return
+        val argReference = argValueExpr.resolveReferenceTarget(bindingContext) ?: return
+
+        if (argReference.fqNameSafe.asString().substringBeforeLast(".") == DISPATCHERS_OBJECT) {
```

Comment:
Maybe we can extract `fqNameSafe.asString().substringBeforeLast(".")` as a function to our helpers? Looks like sth useful